perception/autoware_tensorrt_bevdet/src/bevdet.cpp

```diff
@@ -0,0 +1,717 @@
+  crop_h_ = model_config["data_config"]["crop"][0].as<int>();
+  crop_w_ = model_config["data_config"]["crop"][1].as<int>();
+  mean_[0] = model_config["mean"][0].as<float>();
+  mean_[1] = model_config["mean"][1].as<float>();
+  mean_[2] = model_config["mean"][2].as<float>();
+  std_[0] = model_config["std"][0].as<float>();
+  std_[1] = model_config["std"][1].as<float>();
+  std_[2] = model_config["std"][2].as<float>();
+  down_sample_ = model_config["model"]["down_sample"].as<int>();
+  depth_start_ = model_config["grid_config"]["depth"][0].as<float>();
+  depth_end_ = model_config["grid_config"]["depth"][1].as<float>();
+  depth_step_ = model_config["grid_config"]["depth"][2].as<float>();
+  x_start_ = model_config["grid_config"]["x"][0].as<float>();
+  x_end_ = model_config["grid_config"]["x"][1].as<float>();
+  x_step_ = model_config["grid_config"]["x"][2].as<float>();
+  y_start_ = model_config["grid_config"]["y"][0].as<float>();
+  y_end_ = model_config["grid_config"]["y"][1].as<float>();
+  y_step_ = model_config["grid_config"]["y"][2].as<float>();
+  z_start_ = model_config["grid_config"]["z"][0].as<float>();
+  z_end_ = model_config["grid_config"]["z"][1].as<float>();
+  z_step_ = model_config["grid_config"]["z"][2].as<float>();
+  bevpool_channel_ = model_config["model"]["bevpool_channels"].as<int>();
+  nms_pre_maxnum_ = model_config["test_cfg"]["max_per_img"].as<int>();
+  nms_post_maxnum_ = model_config["test_cfg"]["post_max_size"].as<int>();
+  score_thresh_ = model_config["test_cfg"]["score_threshold"].as<float>();
+  nms_overlap_thresh_ = model_config["test_cfg"]["nms_thr"][0].as<float>();
+  use_depth_ = model_config["use_depth"].as<bool>();
+  use_adj_ = model_config["use_adj"].as<bool>();
+  transform_size_ = model_config["transform_size"].as<int>();
+  cam_params_size_ = model_config["cam_params_size"].as<int>();
+
+  std::vector<std::vector<float>> nms_factor_temp =
+    model_config["test_cfg"]["nms_rescale_factor"].as<std::vector<std::vector<float>>>();
+  nms_rescale_factor_.clear();
+  for (const auto & task_factors : nms_factor_temp) {
+    for (float factor : task_factors) {
+      nms_rescale_factor_.push_back(factor);
+    }
+  }
+
+  std::vector<std::vector<std::string>> class_name_pre_task;
+  class_num_ = 0;
+  YAML::Node tasks = model_config["model"]["tasks"];
+  class_num_pre_task_ = std::vector<int>();
+  for (auto it : tasks) {
+    int num = it["num_class"].as<int>();
+    class_num_pre_task_.push_back(num);
+    class_num_ += num;
+    class_name_pre_task.push_back(it["class_names"].as<std::vector<std::string>>());
+  }
+
+  YAML::Node common_head_channel = model_config["model"]["common_head"]["channels"];
+  YAML::Node common_head_name = model_config["model"]["common_head"]["names"];
+  for (size_t i = 0; i < common_head_channel.size(); i++) {
+    out_num_task_head_[common_head_name[i].as<std::string>()] = common_head_channel[i].as<int>();
+  }
+
```

Comment:
The `out_num_task_head_` map should be a private member, with appropriate accessors to comply with proper encapsulation.  Accessing internal data structures directly from outside the class should be avoided.